packages/catalog-server/src/lib/firestore/firestore-repository.ts

```diff
@@ -234,8 +248,13 @@ export class FirestoreRepository implements Repository {
       const distTags = packageMetadata['dist-tags'];
       const versionDistTags = getDistTagsForVersion(distTags, version);
 
+      const compressedManifest =
+        customElementsManifestSource &&
+        (await brotliCompress(customElementsManifestSource)).toString('base64');
+
       // Store package data and mark version as ready
       t.set(versionRef, {
+        __typename: 'ReadableCompressedPackageVersion',
```

Comment:
Is `__typename` something meaningful in firestore, or just a convention you picked? I see that it's a thing in graphql, but I would assume compression would be handled automatically by the firestore layer, so graphql shouldn't really be relevant?

Reply:
__typename is a GraphQL thing and generated in the schema types by the graphql-codegen tools. It's optional, so we can use those GraphQL types as our internal data model too and generally ignore the __typename field. It'll be set by the GraphQL layers for the RPC calls.

By setting it to a literal that's different from 'ReadablePackageVersion' though, we force the ReadableCompressedPackageVersion to be unassignable to ReadablePackageVersion. I had tried to declare ReadableCompressedPackageVersion.customElementsManifest as `never`, but that didn't work.